core/platform-api/che-core-api-infrastructure-local/src/main/java/org/eclipse/che/api/local/SingleEnvironmentFilter.java

```diff
@@ -99,10 +70,14 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
     private void setWorkspaceIdToContext(EnvironmentContext environmentContext, HttpServletRequest httpRequest) {
         String requestURI = httpRequest.getRequestURI();
 
-        String[] uriParts = requestURI.split("/", 5);
+        String[] uriParts = requestURI.split("/");
+
+        if ((requestURI.startsWith("/ide/ext") || requestURI.startsWith("/ide/api")) && uriParts.length >= 5) {
+            String workspaceId = uriParts[4];
 
-        if (requestURI.contains("/ws") && uriParts.length >= 4) {
-            environmentContext.setWorkspaceId(uriParts[4]);
+            if (workspaceId.startsWith("workspace")) {
```

Comment:
I think requests to /api/worskpace/user123:workspaceName123 or /api/workspace/name/workspaceName123 will be processed wrong